lib/iex/lib/iex/info.ex

```diff
@@ -35,13 +35,27 @@ defimpl IEx.Info, for: Atom do
       end
 
     mod_info = mod.module_info()
-    ["Module bytecode": module_object_file(mod),
-     "Source": module_source_file(mod_info),
-     "Version": module_version(mod_info),
-     "Compile options": module_compile_options(mod_info),
-     "Description": "#{extra}Call #{inspect mod}.module_info() to access metadata.",
-     "Raw representation": ":" <> inspect(Atom.to_string(mod)),
-     "Reference modules": "Module, Atom"]
+    generic_info =
+      ["Module bytecode": module_object_file(mod),
+       "Source": module_source_file(mod_info),
+       "Version": module_version(mod_info),
+       "Compile options": module_compile_options(mod_info),
+       "Description": "#{extra}Call #{inspect mod}.module_info() to access metadata."]
+
+    specific_info =
+      if _protocol? = function_exported?(mod, :__protocol__, 1) do
```

Comment:
Is the module guaranteed to be loaded at this point? Otherwise this check will fail.